tamr_toolbox/utils/version.py

```diff
@@ -49,8 +53,198 @@ def _as_float(version: str) -> float:
+    """
+    assert parse(min_version) < parse(max_version), "min_version must be smaller than max_version"
+    return parse(min_version) <= parse(tamr_version) <= parse(str(max_version))
+
+
+def raise_warn_tamr_version(
+    tamr_version: str, min_version: str, max_version=None, exact_version=False, response="error"
+) -> None:
+    """
+    Check Tamr version and raise error/warn as appropriate.
+
+    Args:
+        tamr_version (str):
+            The version of Tamr being considered
+        min_version (str):
+            The earliest (known) version of Tamr that supports the function
+        max_version (str, optional):
+            The latest (known) version of Tamr that supports the function.
+            Default None, supporting all latest releases of Tamr
+        exact_version (bool, optional):
+            Set to True to only support one release of Tamr. Default is False
+        response (one of ["error", "warn"], optional):
+            Either log a warning and continue on version mismatch, or raise
+            an error and prevent further code execution (default)
+    Notes:
+        If exact_version is True, max_version will be ignored
+
+    See Also:
+        utils.version.ensure_tamr_version
+        utils.version.is_tamr_version_equal
+        utils.version.is_tamr_version_between
+        utils.version.is_tamr_version_atleast
+    """
+
+    allowed_responses = ["error", "warn"]
+    if response not in allowed_responses:
+        raise ValueError(f"Response must be one of {allowed_responses}")
+
+    message = None
+    if exact_version:
+        if not is_tamr_version_equal(tamr_version, min_version):
+            message = (
+                f"Using Tamr version(s) {tamr_version}, " f"but must be exactly {min_version}."
+            )
+
+    elif not max_version:
+        if not is_tamr_version_atleast(tamr_version, min_version):
+            message = (
+                f"Using Tamr version(s) {tamr_version}, " f"but must be at least {min_version}."
+            )
+
+    elif not is_tamr_version_between(tamr_version, min_version, max_version):
+        message = (
+            f"Using Tamr version(s) {tamr_version}, "
+            f"but must be between {min_version} and {max_version}."
+        )
+
+    if message and (response == "error"):
+        raise EnvironmentError(message)
+    elif message and (response == "warn"):
+        warnings.warn(message)
+
+
+def ensure_tamr_version(min_version: str, max_version=None, exact_version=False) -> Callable:
+    """
+    Pie decorator for Tamr version checking
+
+    Args:
+        min_version (str):
+            The earliest (known) version of Tamr that supports the function
+        max_version (str, optional):
+            The latest (known) version of Tamr that supports the function.
+            Default None, supporting all latest releases of Tamr
+        exact_version (bool, optional):
+            Set to True to only support one release of Tamr. Default is False
```

Comment:
Typing in docstrings should be slightly different than this:
```suggestion
def ensure_tamr_version(min_version: str, max_version: Optional[str] = None, exact_version: bool = False) -> Callable:
    """
    Pie decorator for Tamr version checking

    Args:
        min_version: The earliest (known) version of Tamr that supports the function
        max_version: The latest (known) version of Tamr that supports the function.
            Default None, supporting all latest releases of Tamr
        exact_version: Set to True to only support one release of Tamr. Default is False
```